app/models/miq_provision_virt_workflow.rb

```diff
@@ -216,15 +216,19 @@ def available_vlans_and_hosts(options = {})
       rails_logger('allowed_vlans', 0)
       # TODO: Use Active Record to preload this data?
       MiqPreloader.preload(hosts, :switches => :lans)
-      hosts.each do |h|
-        h.lans.each { |l| vlans[l.name] = l.name unless l.switch.shared? }
-      end
+      load_allowed_vlans(hosts, vlans)
       rails_logger('allowed_vlans', 1)
     end
 
     return vlans, hosts
   end
 
+  def load_allowed_vlans(hosts, vlans)
+    hosts.each do |h|
+      h.lans.each { |l| vlans[l.name] = l.name if @vlan_options[:dvs] || !l.switch.shared? }
```

Comment:
Ha so I just noticed you did exactly that here https://github.com/ManageIQ/manageiq-providers-ovirt/pull/27/files#diff-9be4bb7e189cb01956b975459128cc04R96 so completely ignore https://github.com/ManageIQ/manageiq/pull/14991/files#r115488788 :)

Still you should leave the `unless l.switch.shared?` from the original method